Foundation/NSPathUtilities.swift

```diff
@@ -601,9 +601,21 @@ public func NSHomeDirectory() -> String {
 }
 
 public func NSHomeDirectoryForUser(user: String?) -> String? {
```

Comment:
also this might be better shifted to just call out to CFCopyHomeDirectoryURLForUser and any additional logic should be transitioned there